samples/matter/common/src/bridge/bridge_storage_manager.cpp

```diff
@@ -123,31 +200,85 @@ bool BridgeStorageManager::RemoveBridgedDeviceType(uint8_t bridgedDeviceIndex)
 	return Nrf::GetPersistentStorage().NonSecureRemove(&id);
 }
 
-bool BridgeStorageManager::StoreBridgedDevice(const MatterBridgedDevice *device, uint8_t index)
+bool BridgeStorageManager::LoadBridgedDevice(BridgedDevice &device, uint8_t index)
 {
-	if (!device) {
+	Nrf::PersistentStorageNode id = CreateIndexNode(index, &mBridgedDevice);
+	size_t readSize = 0;
+	uint8_t buffer[sizeof(BridgedDevice) + kMaxUserDataSize];
+	uint16_t counter = 0;
+
+	if (!Nrf::PersistentStorage::Instance().Load(&id, buffer, sizeof(BridgedDevice) + kMaxUserDataSize, readSize)) {
 		return false;
 	}
 
-	if (!StoreBridgedDeviceEndpointId(device->GetEndpointId(), index)) {
-		return false;
+	/* Deserialize data and copy it from buffer into structure's fields. */
```

Comment:
Validate `readSize` before reading the buffer?

Reply:
Do you mean validating that readSize is >= than sizeof(BridgedDevice)? I believe that upper limit is already secured, as we pass sizeof buffer, as a max limit to the `Load` method.

Reply:
That `readSize >= sizeof(all mandatory fields)` and then `readSize >= sizeof(all mandatory fields) + label length` before reading the label.

Reply:
Ok I added it and also validation that `readSize >= sizeof(all mandatory fields) +label length + user data size` before reading user data.